extensions/email/src/main/java/org/mule/extension/email/internal/builder/MessageBuilder.java

```diff
@@ -0,0 +1,328 @@
+        return this;
+    }
+
+    /**
+     * Adds attachments represented as a {@link Map} of {@link String}/{@link DataHandler}
+     * to the {@link Message} that is being built.
+     *
+     * @param attachments the attachments that are going to be added to the email.
+     * @return this {@link MessageBuilder}
+     */
+    public MessageBuilder withAttachments(Map<String, DataHandler> attachments)
+    {
+        this.attachments = attachments;
+        return this;
+    }
+
+    /**
+     * Adds the sent date to the {@link Message} that is being built.
+     *
+     * @param date the date in which the email was sent.
+     * @return this {@link MessageBuilder}
+     * @throws MessagingException
+     */
+    public MessageBuilder withSentDate(Date date) throws MessagingException
+    {
+        this.message.setSentDate(date);
+        return this;
+    }
+
+    /**
+     * Adds the text content the {@link Message} that is being built.
+     *
+     * @param content     the text content of the email.
+     * @param contentType the contentType of the {@code content} of the email. One of "text/plain" or "text/html" expected.
+     * @return this {@link MessageBuilder}
+     * @throws MessagingException
+     */
+    public MessageBuilder withContent(String content, String contentType) throws MessagingException
+    {
+        this.content = content;
+        this.contentType = contentType;
+        return this;
+    }
+
+    /**
+     * Adds the text content the {@link Message} that is being built.
+     * The contentType of this content will default to "text/plain".
+     *
+     * @param content the text content of the email.
+     * @return this {@link MessageBuilder}
+     * @throws MessagingException
+     */
+    public MessageBuilder withContent(String content) throws MessagingException
+    {
+        this.content = content;
+        return this;
+    }
+
+    /**
+     * Adds the reply to addresses of the {@link Message} that is being built.
+     *
+     * @param replyAddresses the reply to addresses of the email.
+     * @return this {@link MessageBuilder}
+     * @throws MessagingException
+     */
+    public MessageBuilder replyTo(List<String> replyAddresses) throws MessagingException
+    {
+        this.message.setReplyTo(toAddressArray(replyAddresses));
+        return this;
+    }
+
+    /**
+     * Builds the {@link Message} with all the data provided.
+     *
+     * @return a new {@link MimeMessage} instance.
+     * @throws MessagingException
+     */
+    public MimeMessage build() throws MessagingException
+    {
+        DataHandler contentDataHandler = new DataHandler(content, contentType);
+        if (attachments != null && !attachments.isEmpty())
+        {
+            // first part of the message is the content
+            MimeMultipart multipart = new MimeMultipart();
+            MimeBodyPart mimeBodyPart = new MimeBodyPart();
+            mimeBodyPart.setDisposition(INLINE);
```

Comment:
are you sure the disposition should always be INLINE? It makes sense for some types of images but not sure for things such as documents